js/customize-snapshots.js

```diff
@@ -1,894 +1,971 @@
 
-			if ( api.state( 'snapshot-exists' ).get() ) {
-				api.state( 'saved' ).set( false );
-				component.resetSavedStateQuietly();
+			snapshot.schedule = {};
+
+			if ( _.isObject( snapshotsConfig ) ) {
+				_.extend( snapshot.data, snapshotsConfig );
 			}
 
-			api.trigger( 'snapshots-ready', component );
-		} );
+			// Set the initial client timestamp.
+			snapshot.data.initialClientTimestamp = snapshot.dateValueOf();
 
-		api.bind( 'save', function( request ) {
+			api.bind( 'ready', function() {
+				snapshotExists = '{}' !== api.previewer.query().customized;
+				api.state.create( 'snapshot-exists', snapshotExists );
+				api.state.create( 'snapshot-saved', true );
+				api.state.create( 'snapshot-submitted', true );
 
-			// Make sure that saved state is false so that Published button behaves as expected.
-			api.state( 'saved' ).set( false );
+				snapshot.data.uuid = snapshot.data.uuid || api.settings.changeset.uuid;
+				snapshot.data.title = snapshot.data.title || snapshot.data.uuid;
 
-			request.fail( function( response ) {
-				var id = 'snapshot-dialog-error',
-					snapshotDialogPublishError = wp.template( id );
+				snapshot.editControlSettings = new api.Value( {
+					title: snapshot.data.title,
+					date: snapshot.data.publishDate
+				} );
 
-				if ( response.responseText ) {
+				api.bind( 'change', function() {
+					api.state( 'snapshot-saved' ).set( false );
+					api.state( 'snapshot-submitted' ).set( false );
+				} );
 
-					// Insert the dialog error template.
-					if ( 0 === $( '#' + id ).length ) {
-						$( 'body' ).append( snapshotDialogPublishError( {
-							title: component.data.i18n.publish,
-							message: api.state( 'snapshot-exists' ).get() ? component.data.i18n.permsMsg.update : component.data.i18n.permsMsg.save
-						} ) );
-					}
+				snapshot.frontendPreviewUrl = new api.Value( api.previewer.previewUrl.get() );
+				snapshot.frontendPreviewUrl.link( api.previewer.previewUrl );
 
-					$( '#customize-header-actions .spinner' ).removeClass( 'is-active' );
+				snapshot.addButtons();
+				snapshot.editSnapshotUI();
 
-					// Open the dialog.
-					$( '#' + id ).dialog( {
-						autoOpen: true,
-						modal: true
-					} );
-				}
+				$( '#snapshot-submit' ).on( 'click', function( event ) {
+					event.preventDefault();
+					snapshot.updateSnapshot( 'pending' );
+				} );
+
+				api.trigger( 'snapshots-ready', snapshot );
 			} );
-			return request;
-		} );
 
-		api.bind( 'saved', function( response ) {
-			var url = window.location.href,
-				updatedUrl,
-				urlParts;
+			api.bind( 'save', function( request ) {
 
-			// Update the UUID.
-			if ( response.new_customize_snapshot_uuid ) {
-				component.data.uuid = response.new_customize_snapshot_uuid;
-				component.previewLink.attr( 'target', component.data.uuid );
-			}
+				// Make sure that saved state is false so that Published button behaves as expected.
+				api.state( 'saved' ).set( false ); // @todo Move it to compat.
```

Comment:
That being said, I realize this is part of the original plugin. So yeah, moving to back-compat would make sense.